x/warp/README.md

```diff
@@ -0,0 +1,190 @@
+# Avalanche Warp Messaging
+
+> **Warning**
+> Avalanche Warp Messaging is currently in experimental mode to be used only on ephemeral test networks.
+>
+> Breaking changes to Avalanche Warp Messaging integration into Subnet-EVM may still be made.
+
+Avalanche Warp Messaging offers a basic primitive to enable Cross-Subnet communication on the Avalanche Network.
+
+It is intended to allow communication between arbitrary Custom Virtual Machines (including, but not limited to Subnet-EVM).
+
+## How does Avalanche Warp Messaging Work
+
+Avalanche Warp Messaging relies on the Avalanche P-Chain to provide a read-only view of every Subnet's validator set. In Avalanche, the P-Chain is used to maintain the Primary Network's validator set, create new subnets and blockchains, and maintain the validator sets of each Subnet. As of the Banff Upgrade, Avalanche enables registering a BLS Public Key alongside a validator.
+
+In order to be a validator of an Avalanche Subnet, a node must also validate the Avalanche Primary Network. This means each Subnet validator has read access to the P-Chain state.
+
+With just those two things:
+- Read access to all Subnet validator sets through the P-Chain
+- BLS Public Keys registered on the P-Chain
+
+We can build a generic Avalanche Warp Messaging Protocol. :point_down:
+
+### Subnet to Subnet
+
+The validator set of Subnet A can send a message on behalf of any blockchain on its Subnet.
+
+For example, let's say that there are two Subnets: Subnet A and Subnet B each with a single Blockchain we'll call Blockchain A and Blockchain B.
+
+First, Blockchain A produces a BLS Multisignature of a message to send to Blockchain B:
+
+1. A transaction is issued on Blockchain A to send a message to Subnet B
+2. The transaction is accepted on Blockchain A (must wait for acceptance)
+3. The VM powering Blockchain A (ex: Subnet-EVM) should either
+    a) be willing to sign the message to allow a signature aggregator to aggregate signaures from the Subnet's validator set (existing implementation)
+    b) aggregate signatures from the validator set of Subnet A to produce its own aggregate signature
```

Comment:
can we clarify definition of signature aggregator on line 35

Reply:
Changing to off-chain relayer. I think it's ok to assume an understanding of what an off-chain relayer would do in this context